### PR TITLE
Update Composing Infrastructure doc with ToCompositeFieldPatch example

### PR DESCRIPTION
**Pull request by mcavoyk**
Signed-off-by: Kellin McAvoy <kmcavoy@ea.com>

### Description of your changes

For https://github.com/crossplane/crossplane/issues/2128

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Ran through the example in the doc and updated any related output. Left timestamps equivalent to the original.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/concepts/composition.md [hunk 1]
```diff
@@ -448,6 +454,16 @@ spec:
         - type: math
           math:
             multiply: 1024
+    # Patches can also be applied from the composed resource (MySQLServer)
```
Comment (muvaf) [inline docs/concepts/composition.md:457]: Do they also appear in claim status?

Comment (mcavoyk) [reply docs/concepts/composition.md:457]: Yes, `ToCompositeFieldPath` patches are authoritative for status fields, and can also late initialize spec fields. This applies to both Composites and Claims.

File: docs/concepts/composition.md [hunk 2]
```diff
@@ -328,6 +352,8 @@ spec:
     # object can therefore have unexpected consquences and is not recommended.
     # Instead patch specific annotations by specifying their keys.
     - fromFieldPath: metadata.annotations[example.org/app-name]
+  - name: external-name
```
Comment (mcavoyk) [inline docs/concepts/composition.md:355]: Moved this to a separate patchset so the example XR/XRC's would still work properly.

File: docs/concepts/composition.md (hunk 1 above)
Comment (mcavoyk) [reply docs/concepts/composition.md:482]: Updated comment to clarify that composite is patched and if a claim reference is available, then the claim is also patched.

**Review by hasheddan — APPROVED** [review]
LGTM but I’ll leave final approval to @muvaf. @mcavoyk could you rebase? We will also want to backport this after merge 👍

**Review by muvaf — APPROVED** [review]
Thanks @mcavoyk !

**Comment by mcavoyk** [comment]
@hasheddan PR is rebased

**Comment by hasheddan** [comment]
Thanks @mcavoyk! Can you cherry-pick these 2 commits to `release-1.1`?

**Comment by mcavoyk** [comment]
> Thanks @mcavoyk! Can you cherry-pick these 2 commits to `release-1.1`?

Will do :+1:

**Pull request merged**